src/Storages/System/StorageSystemViewRefreshes.cpp

```diff
@@ -86,7 +86,7 @@ void StorageSystemViewRefreshes::fillData(
 
         res_columns[i++]->insert(refresh.exception_message);
         res_columns[i++]->insert(refresh.refresh_count);
-        res_columns[i++]->insert(Float64(refresh.progress.read_rows) / refresh.progress.total_rows_to_read);
+        res_columns[i++]->insert(std::min(Float64(refresh.progress.read_rows) / refresh.progress.total_rows_to_read, 1.0));
```

Comment:
This is not a good solution - it just hides the problem, not resolves it. The problem is that function `rand64()` produces progress of 1 row, while contributing zero to total. So we have 2 total from `numbers(2)`, zero from `rand64()`, and then during generation we produce progress 2 rows from `numbers(2)` and 1 from `rand64()` - so, progress = generated/total = (2+1)/(2+0)=1.5
The right solution would be either generate total from `rand64()`, or to suppress progress from `rand64()`

Reply:
I think you need to add
```
addTotalRowsApprox(chunk.getNumRows());
```
to `SourceFromSingleChunk` constructors - `SELECT function()` uses it as a source